code/modules/scrap/scrap_beacon.dm

```diff
@@ -0,0 +1,48 @@
+/obj/structure/scrap_beacon
+	name = "Scrap Beacon"
+	desc = "This machine generates directional gravity rays which catch trash orbiting around."
+	icon = 'icons/obj/structures/scrap/scrap_beacon.dmi'
+	icon_state = "beacon0"
+	anchored = TRUE
+	density = TRUE
+	layer = MOB_LAYER + 1
+	var/summon_cooldown = 1200
+	var/impact_speed = 3
+	var/impact_prob = 100
+	var/impact_range = 2
+	var/last_summon = -3000
+	var/active = 0
+
+/obj/structure/scrap_beacon/attack_hand(mob/user)
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+	if((last_summon + summon_cooldown) >= world.time)
+		to_chat(user, "<span class='notice'>[name] not charged yet.</span>")
+		return
+	last_summon = world.time
+	if(!active)
+		start_scrap_summon()
+
+/obj/structure/scrap_beacon/proc/start_scrap_summon()
+	set waitfor = FALSE
+
+	active = TRUE
+	icon_state = "beacon1"
+	sleep(30)
```

Comment:
Timers?

Reply:
sleeps aren't necessarily awful, they're practical for small, simple things like this.

In any case the scrap beacon will not be active content for now